trl/trainer/ppo_config.py

```diff
@@ -128,6 +128,14 @@ class PPOConfig:
     global_batch_size: tyro.conf.Suppress[int] = None
     """TO BE FILLED In RUNTIME: the effective `batch_size` across all processes"""
 
+    if optimize_cuda_cache is not None:
+        warnings.warn(
+            "The `optimize_cuda_cache` arguement will be deprecated soon, please use `optimize_device_cache` instead."
+        )
+        optimize_device_cache = optimize_cuda_cache
+    else:
+        optimize_device_cache = False
+
```

Comment:
Shouldn't we add `optimize_device_cache` already to the config so people can migrate? Can't find it in the config above. cc @younesbelkada

Reply:
Ah yes good point, would you be happy to add that attribute in the config object - i.e. here: https://github.com/abhilash1910/trl/blob/sycl/trl/trainer/ppo_config.py#L102

Reply:
@lvwerra @younesbelkada  added this in the arg list for smoother migration pattern for others to use.  This would also help same apis from other devices -mps, npu etc.